utils/Azure/acrTools.ts

```diff
@@ -0,0 +1,163 @@
+}
+
+//Registry item management
+/** List images under a specific Repository */
+export async function getImagesByRepository(element: Repository): Promise<AzureImage[]> {
+    let allImages: AzureImage[] = [];
+    let image: AzureImage;
+    let tags: string[];
+    const { acrAccessToken } = await acquireACRAccessTokenFromRegistry(element.registry, 'repository:' + element.name + ':pull');
+    await request.get('https://' + element.registry.loginServer + '/v2/' + element.name + '/tags/list', {
+        auth: {
+            bearer: acrAccessToken
+        }
+    }, (err, httpResponse, body) => {
+        if (err) { throw (err) }
+        if (body.length > 0) {
+            tags = JSON.parse(body).tags;
+        }
+    });
+
+    for (let tag of tags) {
+        image = new AzureImage(element, tag);
+        allImages.push(image);
+    }
+    return allImages;
+}
+/** List repositories on a given Registry. */
+export async function getRepositoriesByRegistry(registry: Registry): Promise<Repository[]> {
+    const allRepos: Repository[] = [];
+    let repo: Repository;
+    const { acrRefreshToken, acrAccessToken } = await acquireACRAccessTokenFromRegistry(registry, "registry:catalog:*");
+    await request.get('https://' + registry.loginServer + '/v2/_catalog', {
+        auth: {
+            bearer: acrAccessToken
+        }
+    }, (err, httpResponse, body) => {
+        if (body.length > 0) {
+            const repositories = JSON.parse(body).repositories;
+            for (let tempRepo of repositories) {
+                repo = new Repository(registry, tempRepo, acrAccessToken, acrRefreshToken);
+                allRepos.push(repo);
+            }
+        }
+    });
+
+    //Note these are ordered by default in alphabetical order
+    return allRepos;
+}
+/** Sends a custon html request to a registry
+ * @param http_method : the http method, this function currently only uses delete
+ * @param login_server: the login server of the registry
+ * @param path : the URL path
+ * @param username : registry username, can be in generic form of 0's, used to generate authorization header
+ * @param password : registry password, can be in form of accessToken, used to generate authorization header
+ */
+export async function sendRequestToRegistry(http_method: string, login_server: string, path: string, bearerAccessToken: string): Promise<any> {
+    let url: string = `https://${login_server}${path}`;
+    let header = 'Bearer ' + bearerAccessToken;
+    let opt = {
+        headers: { 'Authorization': header },
+        http_method: http_method,
+        url: url
+    }
+    if (http_method === 'delete') {
+        await request.delete(opt);
+    }
+}
+
+//Credential management
+/** Obtains registry username and password compatible with docker login */
+export async function loginCredentials(registry: Registry): Promise<{ password: string, username: string }> {
```

Comment:
This appears to be unused.

Reply:
It is as of this PR but I included it given that all the other token management items are included in this one. It is used in an upcoming feature we will PR after this one to enable pull.

Reply:
okay.